tests/filecheck/backend/riscv/canonicalize.mlir

```diff
@@ -84,6 +84,12 @@ builtin.module {
   %add_lhs_rhs = riscv.add %i1, %i1 : (!riscv.reg<a1>, !riscv.reg<a1>) -> !riscv.reg<a0>
   "test.op"(%add_lhs_rhs) : (!riscv.reg<a0>) -> ()
 
+  %and_bitwise_zero_l0 = riscv.and %1, 0 : (!riscv.reg<a1>, !riscv.reg<>) -> !riscv.reg<a0>
+  "test.op"(%and_bitwise_zero_l0) : (!riscv.reg<a0>) -> ()
+
+  %and_bitwise_zero_r0 = riscv.and 0, %1 : (!riscv.reg<>, !riscv.reg<a1>) -> !riscv.reg<a0>
+  "test.op"(%and_bitwise_zero_r0) : (!riscv.reg<a0>) -> ()
```

Comment:
```suggestion
  %and_bitwise_zero_l0 = riscv.and %i1, %0 : (!riscv.reg<a1>, !riscv.reg<>) -> !riscv.reg<a0>
  "test.op"(%and_bitwise_zero_l0) : (!riscv.reg<a0>) -> ()

  %and_bitwise_zero_r0 = riscv.and %0, %i1 : (!riscv.reg<>, !riscv.reg<a1>) -> !riscv.reg<a0>
  "test.op"(%and_bitwise_zero_r0) : (!riscv.reg<a0>) -> ()
```
I don't think this can work, since and takes two registers, and not an  immediate value. The constant argument should be %0 for it to work, and the non-constant %i1. I'd recommend running `make filecheck` locally to verify that the test passes, or `lit -v tests/filecheck/backend/riscv/canonicalize.mlir`